interpreter/interpreter.go

```diff
@@ -189,6 +271,11 @@ func (itrp *Interpreter) doVariableAssignment(declaration *semantic.NativeVariab
 		}
 	}
 	scope.Set(declaration.Identifier.Name, value)
+	// If this is a top level variable assignment,
+	// add it to the interpreter's package object.
+	if _, ok := itrp.pkg.types[declaration]; ok {
+		itrp.pkg.value.Set(declaration.Identifier.Name, value)
```

Comment:
Would it be possible to use the scope as the values.Object implementation instead of copying the values into a separate object?

This way the `packageObject` could implment the values.Object interface directly and all it would need is a pointer to the correct top level scope. This would eliminate the need to track which var assignments are top level versus which are not. 

This would mean that above instead of this:

	scope.Set(name, pkg.Value())

You would have this:

	scope.Set(name, pkg)


If you do use the scope though you have to track which identifiers in the scope are public vs private. The only private identifiers would be the imported packages themselves.

Thoughts?

Reply:
I like it. Less memory consumption, and it will greatly simplify the `PolyType()` method. My only concern is like you say keeping track of private vs public identifiers. But I think the benefits outweighs the extra check in the `Get()` method.